denote-dired.el

```diff
@@ -440,21 +422,20 @@ Rename marked files in Dired using the following pattern:
 Batch renaming ignores files that comply with Denote's
 file-naming scheme."
   (interactive nil dired-mode)
-  (if-let ((marks (dired-get-marked-files)))
+  (if-let ((marks (dired-get-marked-files))
+           (keywords (denote--keywords-prompt)))
       (progn
         (dolist (file marks)
-          (let* ((keywords (denote--keywords-prompt))
-                 (dir (file-name-directory file))
-                 (title (file-name-sans-extension (file-name-nondirectory file)))
+          (let* ((dir (file-name-directory file))
+                 (id (denote-dired--file-name-id file))
+                 (title (or (denote-retrieve--value-title file)
+                            (file-name-sans-extension
+                             (file-name-nondirectory file))))
                  (extension (file-name-extension file t))
                  (new-name (denote--format-file
-                            dir
-                            (denote-dired--file-name-id file)
-                            keywords
-                            (denote--sluggify title)
-                            extension)))
-            (unless (denote--only-note-p file)
-              (rename-file (file-name-nondirectory file) new-name))))
+                            dir id keywords (denote--sluggify title) extension)))
+            (when (denote--only-note-p file)
```

Comment:
The `when` instead of `unless` changes the effect here, no?  The original plan was to ignore files that already conform with the file-naming scheme

Reply:
Yes, I just noticed as well and I pushed a commit to remove the condition. I don't think we need it. It will handle Denote notes correctly, I think. Only the keywords will be changed in this case.

Reply:
Okay.  Yesterday when I was trying it, I had to add the `unless` otherwise it kept renaming the file.

Reply:
It is working fine now?